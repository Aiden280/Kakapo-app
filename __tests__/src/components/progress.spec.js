import React from 'react';
import { shallow } from 'enzyme';
import Progress from 'components/ui/progress/progress';

function setup(props = {}) {
  return { props, wrapper: shallow(<Progress {...props} />) };
}

test('<Progress/> render', () => {
  const { wrapper } = setup();
  expect(wrapper.type()).toBe('div');
  expect(wrapper.prop('className')).toBe('progress');
});

test('<Progress/> rounds 0.415 to 42% ', () => {
  const { wrapper } = setup({ progress: 0.415 });
  expect(wrapper.find('.progress-text').text()).toBe('42%');
});
