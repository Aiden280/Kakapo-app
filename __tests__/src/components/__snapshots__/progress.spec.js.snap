// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Progress/> render 1`] = `
<div
  className="progress"
>
  <div
    className="progress-barberpole"
    style={
      Object {
        "width": "NaN%",
      }
    }
  />
  <div
    className="progress-text"
  >
    NaN
    %
  </div>
</div>
`;

exports[`<Progress/> rounds 0.415 to 42%  1`] = `
<div
  className="progress"
>
  <div
    className="progress-barberpole"
    style={
      Object {
        "width": "42%",
      }
    }
  />
  <div
    className="progress-text"
  >
    42
    %
  </div>
</div>
`;
